java/core/lance-jni/src/blocking_scanner.rs

```diff
@@ -53,7 +54,21 @@ impl BlockingScanner {
 
     pub fn schema(&self) -> Result<SchemaRef> {
         let res = RT.block_on(self.inner.schema())?;
-        Ok(res)
+        let mut new_fields = Vec::new();
+        for field in res.clone().fields() {
+            if field.name() == ROW_ID {
+                let new_field = match field.data_type() {
+                    DataType::UInt64 => {
```

Comment:
Trying to understand why making this change
 Arrow in Java also have DataType:UInt64 but Spark cannot convert UInt64 to SparkType?

Could we modify the Spark Type to Arrow Type conversion directly? Like having an extension??